pkg/auth/providers/ldap/ldap_client.go

```diff
@@ -166,6 +167,58 @@ func (p *ldapProvider) getPrincipalsFromSearchResult(result *ldapv2.SearchResult
 		}
 	}
 
+	allowedPrincipalIDs := config.AllowedPrincipalIDs
+	for _, allowedPid := range allowedPrincipalIDs {
+		parts := strings.SplitN(allowedPid, ":", 2)
+		if len(parts) != 2 {
+			return userPrincipal, groupPrincipals, errors.Errorf("invalid id %v", allowedPid)
+		}
+		scope := parts[0]
+		userType := strings.Split(scope, "_")[1]
```

Comment:
I believe the value is either "user" or "group"? if yes rename variable to entityType?